packages/palette/src/elements/DonutChart/DonutChart.tsx

```diff
@@ -0,0 +1,164 @@
+import React, { useRef, useState } from "react"
+import { Spring } from "react-spring"
+import styled from "styled-components"
+import { color, space } from "../../helpers"
+import { Color } from "../../Theme"
+import { Box } from "../Box"
+import { ChartHoverTooltip } from "../DataVis/ChartHoverTooltip"
+import { coerceTooltip } from "../DataVis/ChartTooltip"
+import { ProvideMousePosition } from "../DataVis/MousePositionContext"
+import { ChartProps } from "../DataVis/utils/SharedTypes"
+import { useGetWrapperWidth } from "../DataVis/utils/useGetWrapperWidth"
+import { useHasEnteredViewport } from "../DataVis/utils/useHasEnteredViewPort"
+import { Sans } from "../Typography"
+
+const colors: Color[] = ["black10", "black30", "black60"]
+
+export interface DonutChartProps extends ChartProps {
+  margin?: number
+}
+
+/**
+ * DonutChart is a component that displays data points with donut shaped arcs.
+ * Good for illustrating numerical proportions.
+ */
+export const DonutChart: React.FC<DonutChartProps> = ({
+  points,
+  margin = space(3),
+}: DonutChartProps) => {
+  const [hoverIndex, setHoverIndex] = useState(-1)
+  const [rest, setRest] = useState(false)
+
+  const wrapperRef = useRef<HTMLDivElement>(null)
+
+  const hasEnteredViewport = useHasEnteredViewport(wrapperRef)
+
+  const width = useGetWrapperWidth(wrapperRef)
+
+  const donutWidth = width * 0.12
+  const donutLabelMargin = space(2) / Math.sqrt(2)
+
+  const centerX = width / 2 + margin
+  const centerY = width / 2 + margin
+
+  const values = points.map(d => d.value).sort()
+  const angelInterpolator = interpolate(0, 2 * Math.PI)
+
+  const arc = d3_arc()
+    .innerRadius(width / 2 - margin - donutWidth)
+    .outerRadius(width / 2 - margin)
+    .padAngle(Math.PI / 360)
+
+  // virtual donut to use its arc centroids to calculate label position
+  const labelArc = d3_arc()
+    .innerRadius(width / 2 - margin + donutLabelMargin)
+    .outerRadius(width / 2 - margin + donutLabelMargin)
+    .padAngle(Math.PI / 360)
+
+  const pie = d3_pie()
+
+  const tooltip = points.map((point, index) => {
+    const hover = hoverIndex === index
+    return hover ? (
+      <ChartHoverTooltip key={`${index}-hover`}>
+        {coerceTooltip(point.tooltip)}
+      </ChartHoverTooltip>
+    ) : null
+  })
+
+  const labels = pie(values).map((slice, index) => {
+    const [x, y] = labelArc.centroid(slice as any)
+    const axisLabelX = points[index].axisLabelX
+    return (
+      axisLabelX && (
+        <DonutLabelContainer
+          key={index}
+          opacity={rest ? 1 : 0}
+          x={x + centerX}
+          y={y + centerY}
+          center={centerX}
+        >
+          <Sans color="black60" size="2">
+            {coerceTooltip(axisLabelX)}
+          </Sans>
+        </DonutLabelContainer>
+      )
+    )
+  })
+
+  const svg = (
+    <svg key={"svg"} width={width + margin} height={width + margin}>
+      <g transform={`translate(${centerX}, ${centerY})`}>
```

Comment:
According to the spec, the donut is supposed to have a silhouette ring visible before the arcs animate in.

![image](https://user-images.githubusercontent.com/1242537/59096946-313ddd80-8914-11e9-93b6-0dd7f26326f9.png)

Reply:
Added here: https://github.com/artsy/palette/pull/495/commits/1e08b48840a6d8747392def6069bbf495b89cb83

Reply:
Looks great, thanks!